### ISSUE TITLE
Landscape mode C#

### ISSUE THREAD

**Issue by cavaIV** [ios-driver/ios-driver#250]
Hey guys!

Is it possible to set the iOS Simulator to the landscape mode with C#?
If it is possible, how? :)

**Comment by freynaud** [comment]
the regular landscape and portrait from the selenium api should work. Is that not the case ?

The only ios specific orientation that require sending custom orientation strings are portrait upsidedown and lanscape left

**Comment by cavaIV** [comment]
No the landscape and portrait mode from the selenium api do not work.

Is there another way to change the orientation?
Maybe with JavaScript or something?

**Comment by freynaud** [comment]
Can you let me know the error you see ?
https://github.com/ios-driver/ios-driver/blob/dev/server/src/test/java/org/uiautomation/ios/e2e/config/OrientationTest.java#L44
is using the selenium api, and it passes.

**Comment by cavaIV** [comment]
There's no Augmenter in C#

**Comment by freynaud** [comment]
It's part of the selenium protocol, so it should be implemented.

Anyhow, if you don't have the method in C#, you can execute raw automation JS.
To change the orientation, you can try ;

  driver.executeScript("UIATarget.localTarget().setDeviceOrientation(UIA_DEVICE_ORIENTATION_LANDSCAPELEFT);");

List of orientation : 
https://github.com/ios-driver/ios-driver/blob/dev/common/src/main/java/org/uiautomation/ios/UIAModels/Orientation.java

**Comment by cavaIV** [comment]
If i try the javascript command i get the following error:

An exception of type 'System.InvalidOperationException' occurred in WebDriver.dll but was not handled in user code

Additional information: JS error :ReferenceError: Can't find variable: UIATarget

**Comment by freynaud** [comment]
what app are you using ? if it's mobile safari, you need to switch to the
Native window first,otherwise you execute JS in the browser.

**Comment by cavaIV** [comment]
Thanks @freynaud !
It's working.
But if i change the orientation and open a few HTML5 games with my test a java error occurs:

http://i59.tinypic.com/1zg84jl.png

after that ios-driver shows the following output:

---

 A fatal error has been detected by the Java Runtime Environment:

  SIGSEGV (0xb) at pc=0x00007fff806a58b5, pid=450, tid=370455

 JRE version: 7.0_25-b15
 Java VM: Java HotSpot(TM) 64-Bit Server VM (23.25-b01 mixed mode bsd-amd64 compressed oops)
 Problematic frame:
 C  [libGL.dylib+0x18b5]  glGetString+0xf

 Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again

 An error report file with more information is saved as:
 /Users/localadmin/hs_err_pid450.log

 If you would like to submit a bug report, please visit:
   http://bugreport.sun.com/bugreport/crash.jsp
 The crash happened outside the Java Virtual Machine in native code.
 See problematic frame for where to report the bug.

##  /Users/localadmin/Desktop/StartiOSDriver: line 2:   450 Abort trap: 6           java -jar ~/Desktop/ios-server-standalone-0.6.6-SNAPSHOT.jar -port 4444

Could it be possible that this is cause by the ios-server.jar?

**Comment by freynaud** [comment]
glad it worked. Can you open a new bug with the new issue, and attach hs_err_pid450.log ?

thanks
françois

**Comment by ghost** [comment]
@cavaIV : Could you please share details on the approach you took to "switch to a native window first" before executing the JS command on the driver instance?

Were you opening a native app first, running the JS  to switch the orientation to landscape and then launching mobile safari within the simulator? 

Thanks
Vinay

**Comment by cavaIV** [comment]
I start the simulator, start mobile safari and then i rotate the simulator with the following commands:

var orientationString = "UIA_DEVICE_ORIENTATION_LANDSCAPELEFT";
Driver.SwitchTo().Window("Native");
((IJavaScriptExecutor)this.Driver).ExecuteScript("UIATarget.localTarget().setDeviceOrientation(" + orientationString + ");");
Driver.SwitchTo().Window("Web");

**Comment by ghost** [comment]
@cavaIV :that worked perfectly. Thanks for sharing.
